validator/client/validator.go

```diff
@@ -369,6 +370,85 @@ func (v *validator) SlotDeadline(slot types.Slot) time.Time {
 	return time.Unix(int64(v.genesisTime), 0 /*ns*/).Add(secs * time.Second)
 }
 
+// CheckDoppelGanger checks if the current actively provided keys have
+// any duplicates active in the network.
+func (v *validator) CheckDoppelGanger(ctx context.Context) error {
+	if !featureconfig.Get().EnableDoppleGanger {
+		return nil
+	}
+	log.Info("Running Doppleganger Check")
+	pubkeys, err := v.keyManager.FetchValidatingPublicKeys(ctx)
+	if err != nil {
+		return err
+	}
+	// Exit early if no validating pub keys are found.
+	if len(pubkeys) == 0 {
+		return nil
+	}
+	req := &ethpb.DoppelGangerRequest{ValidatorRequests: []*ethpb.DoppelGangerRequest_ValidatorRequest{}}
+	for _, pkey := range pubkeys {
+		attRec, err := v.db.AttestationHistoryForPubKey(ctx, pkey)
+		if err != nil {
+			return err
+		}
+		if len(attRec) == 0 {
+			// If no history exists we simply send in a zero
+			// value for the request type.
+			req.ValidatorRequests = append(req.ValidatorRequests,
+				&ethpb.DoppelGangerRequest_ValidatorRequest{
+					PublicKey:  pkey[:],
+					Epoch:      0,
+					SignedRoot: make([]byte, 32),
+				})
+			continue
+		}
+		r := retrieveLatestRecord(attRec)
+		req.ValidatorRequests = append(req.ValidatorRequests,
+			&ethpb.DoppelGangerRequest_ValidatorRequest{
+				PublicKey:  r.PubKey[:],
+				Epoch:      r.Target,
+				SignedRoot: r.SigningRoot[:],
+			})
+	}
+	resp, err := v.validatorClient.CheckDoppelGanger(ctx, req)
+	if err != nil {
+		return err
+	}
+	// If nothing is returned by the beacon node, we can exit early.
+	if resp == nil || resp.Responses == nil || len(resp.Responses) == 0 {
+		return nil
+	}
+	for _, valRes := range resp.Responses {
+		if valRes.DuplicateExists {
+			return errors.Errorf("Duplicate instance exists in the network for validator %#x", valRes.PublicKey)
+		}
+	}
```

Comment:
Optional: check all of the validators and build an error with all of the duplicates found instead of returning on just the first one.